### PR TITLE
CI: Build python and java artifacts in travis.ci

### PR DESCRIPTION
**Pull request by charliestrawn-wf**
@Workiva/messaging-pp this PR builds all of the libraries except for dart (need thrift on pub for that) and also runs the compiler tests.

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
<details>
<summary>Number of Findings: 0</summary>

</details>

File: scripts/travis/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+if [ "$TRAVIS_BRANCH" = 'master' ] && [ "$TRAVIS_PULL_REQUEST" == 'false' ]; then
+    mvn deploy -P sign,build-extras --settings mvnsettings.xml
```
Comment (brianshannan-wf) [inline scripts/travis/deploy.sh]: Is this where we will need to publish to pypi in the future?

Comment (charliestrawn-wf) [inline scripts/travis/deploy.sh]: yep, this isn't being called yet. Will follow up with pushing the artifacts in the next PR

**Comment by brianshannan-wf** [comment]
+1

**Pull request merged**